### PR TITLE
autogenerated-apimanagementapidiagnostic

### PR DESCRIPTION
**Pull request by zikalino**

### CODE REVIEW

File: list_resources.sh [hunk 1]
```diff
@@ -4,7 +4,7 @@ cd `dirname $0`
 for file in `ls ./specs`; do
 	work_dir=./specs/$file
 	if [ -d $work_dir ]; then
-		if [ -n "$(git status -s $work_dir)" ];
+		if [ -n "$(git diff --name-only remotes/origin/master remotes/origin/$source_branch -- $work_dir)" ];
```
Comment (metacpp) [inline list_resources.sh]: What if we need to run it in local environment ? Can we add a fallback logic ?

File: azure-pipelines.yml [hunk 2]
```diff
@@ -79,6 +79,7 @@ steps:
   failOnStderr: true
 
 - bash: |
+    export source_branch=$(System.PullRequest.SourceBranch)
```
Comment (metacpp) [inline azure-pipelines.yml]: I would recommend pass the `source_branch` as argument to Makefile instead of using environment variable.

Comment (houkms) [reply azure-pipelines.yml]: OK. Now the `source_branch` is passed as argument.

File: list_resources.sh (hunk 1 above)
Comment (houkms) [reply list_resources.sh]: Add the fallback logic in script.

- We can run `make build SOURCE_BRANCH=branchname`, to generate codes for new added/modified resources in local branch 
- If `SOURCE_BRANCH` is not set, it uses current branch as default, and generates codes for new added/modified resources in current branch
- For uncommitted changes, just uses `make build RESOURCE=resourcename` as before

**Comment by houkms** [comment]
Codes are modified according to the comments.

**Pull request merged**